// apps-mobile/App.tsx
import { useEffect } from 'react';
import { View, StyleSheet } from 'react-native';
import { StatusBar } from 'expo-status-bar';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useColorScheme } from 'react-native';
import { useAppStore } from './src/store/useAppStore';
import { AppHeader } from './src/components/AppHeader';

import { HomeScreen } from './src/screens/HomeScreen';
import { VehicleScreen } from './src/screens/VehicleScreen';
import { RideInputScreen } from './src/screens/RideInputScreen';
import { WorthItScreen } from './src/screens/WorthItScreen';
import { DailyGoalScreen } from './src/screens/DailyGoalScreen';
import { HistoryScreen } from './src/screens/HistoryScreen';
import { SettingsScreen } from './src/screens/SettingsScreen';

const Stack = createNativeStackNavigator();
const Tab = createBottomTabNavigator();

function MainTabs() {
  return (
    <Tab.Navigator
      screenOptions={{
        headerShown: false,
        tabBarStyle: { minHeight: 56 },
        tabBarLabelStyle: { fontSize: 12 },
        tabBarActiveTintColor: '#2563eb',
        tabBarInactiveTintColor: '#64748b',
      }}
    >
      <Tab.Screen name="Home" component={HomeScreen} options={{ title: 'Início' }} />
      <Tab.Screen name="WorthIt" component={WorthItScreen} options={{ title: 'Vale a pena?' }} />
      <Tab.Screen name="Ride" component={RideInputScreen} options={{ title: 'Corrida' }} />
      <Tab.Screen name="Goal" component={DailyGoalScreen} options={{ title: 'Meta' }} />
      <Tab.Screen name="History" component={HistoryScreen} options={{ title: 'Histórico' }} />
    </Tab.Navigator>
  );
}

function MainScreen() {
  return (
    <View style={styles.main}>
      <AppHeader />
      <View style={styles.tabs}>
        <MainTabs />
      </View>
    </View>
  );
}

export default function App() {
  const hydrate = useAppStore((s) => s.hydrate);
  const hydrated = useAppStore((s) => s.hydrated);

  useEffect(() => {
    hydrate();
  }, [hydrate]);

  const colorScheme = useColorScheme();
  const theme = useAppStore((s) => s.settings.theme);
  const isDark = theme === 'dark' || (theme === 'system' && colorScheme === 'dark');

  if (!hydrated) {
    return null;
  }

  return (
    <NavigationContainer>
      <StatusBar style={isDark ? 'light' : 'dark'} />
      <Stack.Navigator screenOptions={{ headerShown: false }}>
        <Stack.Screen name="Main" component={MainScreen} />
        <Stack.Screen name="Vehicle" component={VehicleScreen} />
        <Stack.Screen name="Settings" component={SettingsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  main: {
    flex: 1,
  },
  tabs: {
    flex: 1,
  },
});
