// apps-mobile/src/store/useAppStore.ts
import { create } from 'zustand';
import {
  type VehicleProfile,
  type RideRecord,
  type AppSettings,
  type RideInput,
  type WorthItThresholds,
  type DailyGoal,
  calculateRideCost,
  getDateKey,
  ridesToDailySummaries,
  netProfitForDate,
  averageNetProfitPerRide,
  DEFAULT_APP_SETTINGS,
  DEFAULT_WORTH_IT_THRESHOLDS,
  DEFAULT_DAILY_GOAL,
} from '@driver-profit/shared';
import { storage } from '../storage/asyncStorage';

const VEHICLES_KEY = 'vehicles';
const RIDES_KEY = 'rides';
const SETTINGS_KEY = 'settings';

function defaultSettings(): AppSettings {
  return {
    ...DEFAULT_APP_SETTINGS,
    vehicleId: null,
    dailyGoal: DEFAULT_DAILY_GOAL,
    worthItThresholds: DEFAULT_WORTH_IT_THRESHOLDS,
    currency: 'BRL',
    theme: 'system',
  };
}

interface AppState {
  hydrated: boolean;
  vehicles: VehicleProfile[];
  rides: RideRecord[];
  settings: AppSettings;

  setVehicles: (v: VehicleProfile[]) => void;
  addVehicle: (v: Omit<VehicleProfile, 'id' | 'createdAt'>) => void;
  updateVehicle: (id: string, v: Partial<VehicleProfile>) => void;
  deleteVehicle: (id: string) => void;
  getVehicle: (id: string) => VehicleProfile | undefined;

  setRides: (r: RideRecord[]) => void;
  addRide: (input: RideInput) => RideRecord | null;
  deleteRide: (id: string) => void;

  setSettings: (s: Partial<AppSettings>) => void;
  setWorthItThresholds: (t: WorthItThresholds) => void;
  setDailyGoal: (g: DailyGoal) => void;

  persist: () => void;
  hydrate: () => Promise<void>;

  currentVehicle: () => VehicleProfile | null;
  todayDateKey: () => string;
  todayNetProfit: () => number;
  dailySummaries: () => ReturnType<typeof ridesToDailySummaries>;
  ridesForDate: (dateKey: string) => RideRecord[];
  averageProfitPerRide: () => number;
  ridesNeededForGoal: () => number;
}

function generateId(): string {
  return Date.now().toString(36) + Math.random().toString(36).slice(2);
}

export const useAppStore = create<AppState>((set, get) => ({
  hydrated: false,
  vehicles: [],
  rides: [],
  settings: defaultSettings(),

  setVehicles: (vehicles) => set({ vehicles }),

  addVehicle: (v) => {
    const vehicle: VehicleProfile = {
      ...v,
      id: generateId(),
      createdAt: Date.now(),
    };
    set((s) => {
      const isFirst = s.vehicles.length === 0;
      return {
        vehicles: [...s.vehicles, vehicle],
        settings: isFirst ? { ...s.settings, vehicleId: vehicle.id } : s.settings,
      };
    });
  },

  updateVehicle: (id, patch) => {
    set((s) => ({
      vehicles: s.vehicles.map((v) => (v.id === id ? { ...v, ...patch } : v)),
    }));
  },

  deleteVehicle: (id) => {
    set((s) => ({
      vehicles: s.vehicles.filter((v) => v.id !== id),
      settings:
        s.settings.vehicleId === id
          ? { ...s.settings, vehicleId: s.vehicles[0]?.id ?? null }
          : s.settings,
    }));
  },

  getVehicle: (id) => get().vehicles.find((v) => v.id === id),

  setRides: (rides) => set({ rides }),

  addRide: (input) => {
    const vehicle = get().currentVehicle();
    if (!vehicle) return null;
    const breakdown = calculateRideCost(input, vehicle);
    const now = Date.now();
    const record: RideRecord = {
      ...input,
      ...breakdown,
      id: generateId(),
      vehicleId: vehicle.id,
      timestamp: now,
      dateKey: getDateKey(now),
    };
    set((s) => ({ rides: [record, ...s.rides] }));
    return record;
  },

  deleteRide: (id) => set((s) => ({ rides: s.rides.filter((r) => r.id !== id) })),

  setSettings: (patch) =>
    set((s) => ({ settings: { ...s.settings, ...patch } })),

  setWorthItThresholds: (worthItThresholds) =>
    set((s) => ({ settings: { ...s.settings, worthItThresholds } })),

  setDailyGoal: (dailyGoal) =>
    set((s) => ({ settings: { ...s.settings, dailyGoal } })),

  persist: () => {
    const { vehicles, rides, settings } = get();
    storage.set(VEHICLES_KEY, vehicles);
    storage.set(RIDES_KEY, rides);
    storage.set(SETTINGS_KEY, settings);
  },

  hydrate: async () => {
    const [vehicles, rides, savedSettings] = await Promise.all([
      storage.get<VehicleProfile[]>(VEHICLES_KEY),
      storage.get<RideRecord[]>(RIDES_KEY),
      storage.get<AppSettings>(SETTINGS_KEY),
    ]);
    set({
      hydrated: true,
      vehicles: vehicles ?? [],
      rides: rides ?? [],
      settings: {
        ...defaultSettings(),
        vehicleId: savedSettings?.vehicleId ?? null,
        dailyGoal: savedSettings?.dailyGoal ?? DEFAULT_DAILY_GOAL,
        worthItThresholds: savedSettings?.worthItThresholds ?? DEFAULT_WORTH_IT_THRESHOLDS,
        currency: savedSettings?.currency ?? 'BRL',
        theme: savedSettings?.theme ?? 'system',
      },
    });
  },

  currentVehicle: () => {
    const { vehicles, settings } = get();
    if (!settings.vehicleId) return null;
    return vehicles.find((v) => v.id === settings.vehicleId) ?? null;
  },

  todayDateKey: () => getDateKey(Date.now()),

  todayNetProfit: () => {
    const { rides, todayDateKey } = get();
    return netProfitForDate(rides, todayDateKey());
  },

  dailySummaries: () => ridesToDailySummaries(get().rides),

  ridesForDate: (dateKey) => get().rides.filter((r) => r.dateKey === dateKey),

  averageProfitPerRide: () => averageNetProfitPerRide(get().rides),

  ridesNeededForGoal: () => {
    const { todayNetProfit, settings, averageProfitPerRide } = get();
    const remaining = settings.dailyGoal.targetNetProfit - todayNetProfit();
    if (remaining <= 0) return 0;
    const avg = averageProfitPerRide();
    if (avg <= 0) return 0;
    return Math.ceil(remaining / avg);
  },
}));

useAppStore.subscribe(() => useAppStore.getState().persist());
