// apps-mobile/src/components/AppHeader.tsx
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';

export function AppHeader() {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Driver Profit</Text>
      <View style={styles.actions}>
        <TouchableOpacity
          onPress={() => (navigation as any).navigate('Vehicle')}
          style={styles.iconButton}
          accessibilityLabel="Veículo"
        >
          <Text style={styles.icon}>🚗</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => (navigation as any).navigate('Settings')}
          style={styles.iconButton}
          accessibilityLabel="Configurações"
        >
          <Text style={styles.icon}>⚙️</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#e2e8f0',
  },
  title: {
    fontSize: 18,
    fontWeight: '600',
    color: '#0f172a',
  },
  actions: {
    flexDirection: 'row',
    gap: 8,
  },
  iconButton: {
    padding: 8,
    minWidth: 44,
    minHeight: 44,
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    fontSize: 20,
  },
});
