// apps-mobile/src/screens/DailyGoalScreen.tsx
import { ScrollView, View, Text, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useAppStore } from '../store/useAppStore';
import { formatCurrency } from '../lib/format';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';

export function DailyGoalScreen() {
  const navigation = useNavigation();
  const todayNetProfit = useAppStore((s) => s.todayNetProfit());
  const dailyGoal = useAppStore((s) => s.settings.dailyGoal);
  const ridesNeededForGoal = useAppStore((s) => s.ridesNeededForGoal());
  const averageProfitPerRide = useAppStore((s) => s.averageProfitPerRide());
  const currency = useAppStore((s) => s.settings.currency);

  const remaining = Math.max(0, dailyGoal.targetNetProfit - todayNetProfit);
  const progress =
    dailyGoal.targetNetProfit > 0
      ? Math.min(1, todayNetProfit / dailyGoal.targetNetProfit)
      : 0;

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <View style={styles.section}>
        <Text style={styles.label}>Meta do dia</Text>
        <Text style={styles.bigAmount}>
          {formatCurrency(dailyGoal.targetNetProfit, currency)}
        </Text>
      </View>

      <Card>
        <CardHeader title="Progresso" />
        <View style={styles.cardContent}>
          <View style={styles.progressBar}>
            <View style={[styles.progressFill, { width: `${progress * 100}%` }]} />
          </View>
          <View style={styles.grid}>
            <View style={styles.gridItem}>
              <Text style={styles.gridValue}>
                {formatCurrency(todayNetProfit, currency)}
              </Text>
              <Text style={styles.gridLabel}>Lucro hoje</Text>
            </View>
            <View style={styles.gridItem}>
              <Text style={styles.gridValue}>
                {formatCurrency(remaining, currency)}
              </Text>
              <Text style={styles.gridLabel}>Faltando</Text>
            </View>
          </View>
          {remaining <= 0 && (
            <Text style={styles.success}>Meta atingida!</Text>
          )}
          {remaining > 0 && averageProfitPerRide > 0 && (
            <Text style={styles.hint}>
              Aproximadamente <Text style={styles.bold}>{ridesNeededForGoal}</Text> corrida(s) para a meta
              (média {formatCurrency(averageProfitPerRide, currency)}/corrida)
            </Text>
          )}
          <Button
            variant="secondary"
            fullWidth
            onPress={() => (navigation as any).navigate('Settings')}
          >
            Alterar meta nas configurações
          </Button>
        </View>
      </Card>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  section: { marginBottom: 24 },
  label: { fontSize: 14, color: '#64748b', marginBottom: 4 },
  bigAmount: { fontSize: 24, fontWeight: '700', color: '#0f172a' },
  cardContent: { padding: 16 },
  progressBar: {
    height: 16,
    backgroundColor: '#e2e8f0',
    borderRadius: 8,
    overflow: 'hidden',
    marginBottom: 16,
  },
  progressFill: {
    height: '100%',
    backgroundColor: '#22c55e',
    borderRadius: 8,
  },
  grid: { flexDirection: 'row', gap: 16, marginBottom: 16 },
  gridItem: { flex: 1, alignItems: 'center' },
  gridValue: { fontSize: 20, fontWeight: '600', color: '#0f172a' },
  gridLabel: { fontSize: 12, color: '#64748b', marginTop: 4 },
  success: { color: '#16a34a', fontWeight: '600', marginBottom: 16, textAlign: 'center' },
  hint: { fontSize: 14, color: '#475569', marginBottom: 16, textAlign: 'center' },
  bold: { fontWeight: '700' },
});
