// apps-mobile/src/screens/HistoryScreen.tsx
import { useState } from 'react';
import { ScrollView, View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useAppStore } from '../store/useAppStore';
import { formatCurrency, formatKm } from '../lib/format';
import { Card, CardHeader } from '../components/Card';

function formatDateKey(dateKey: string): string {
  const [y, m, d] = dateKey.split('-');
  const date = new Date(parseInt(y, 10), parseInt(m, 10) - 1, parseInt(d, 10));
  return date.toLocaleDateString('pt-BR', {
    weekday: 'short',
    day: 'numeric',
    month: 'short',
  });
}

export function HistoryScreen() {
  const dailySummaries = useAppStore((s) => s.dailySummaries());
  const ridesForDate = useAppStore((s) => s.ridesForDate);
  const currency = useAppStore((s) => s.settings.currency);
  const [expandedDate, setExpandedDate] = useState<string | null>(null);

  if (dailySummaries.length === 0) {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.content}>
        <Card>
          <View style={styles.empty}>
            <Text style={styles.emptyText}>
              Nenhuma corrida registrada ainda. Registre sua primeira corrida em Corrida.
            </Text>
          </View>
        </Card>
      </ScrollView>
    );
  }

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <Card>
        <CardHeader title="Resumo por dia" />
        {dailySummaries.map((s) => {
          const isExpanded = expandedDate === s.dateKey;
          const rides = ridesForDate(s.dateKey);
          return (
            <View key={s.dateKey}>
              <TouchableOpacity
                style={styles.summaryRow}
                onPress={() => setExpandedDate(isExpanded ? null : s.dateKey)}
                activeOpacity={0.7}
              >
                <View>
                  <Text style={styles.summaryDate}>{formatDateKey(s.dateKey)}</Text>
                  <Text style={styles.summaryMeta}>
                    {s.totalRides} corrida(s) · {formatKm(s.totalDistanceKm)}
                  </Text>
                </View>
                <Text
                  style={[
                    styles.summaryProfit,
                    s.netProfit >= 0 ? styles.positive : styles.negative,
                  ]}
                >
                  {formatCurrency(s.netProfit, currency)}
                </Text>
              </TouchableOpacity>
              {isExpanded &&
                rides.map((r) => (
                  <View key={r.id} style={styles.rideRow}>
                    <Text style={styles.rideLabel}>
                      {formatKm(r.distanceKm)} ·{' '}
                      {r.rideType === 'delivery' ? 'Entrega' : 'Passageiro'}
                    </Text>
                    <Text
                      style={[
                        styles.rideValue,
                        r.netProfit >= 0 ? styles.positive : styles.negative,
                      ]}
                    >
                      {formatCurrency(r.netProfit, currency)}
                    </Text>
                  </View>
                ))}
            </View>
          );
        })}
      </Card>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  empty: { padding: 24, alignItems: 'center' },
  emptyText: { fontSize: 15, color: '#64748b', textAlign: 'center' },
  summaryRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 14,
    borderTopWidth: 1,
    borderTopColor: '#f1f5f9',
  },
  summaryDate: { fontWeight: '600', color: '#0f172a', fontSize: 16 },
  summaryMeta: { fontSize: 14, color: '#64748b', marginTop: 2 },
  summaryProfit: { fontWeight: '600', fontSize: 16 },
  rideRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 10,
    paddingLeft: 32,
    backgroundColor: '#f8fafc',
  },
  rideLabel: { fontSize: 14, color: '#475569' },
  rideValue: { fontSize: 14, fontWeight: '500' },
  positive: { color: '#16a34a' },
  negative: { color: '#dc2626' },
});
