// apps-mobile/src/screens/HomeScreen.tsx
import { ScrollView, View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useAppStore } from '../store/useAppStore';
import { formatCurrency, formatKm } from '../lib/format';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';

export function HomeScreen() {
  const navigation = useNavigation();
  const vehicle = useAppStore((s) => s.currentVehicle());
  const todayNetProfit = useAppStore((s) => s.todayNetProfit());
  const dailyGoal = useAppStore((s) => s.settings.dailyGoal);
  const todayDateKey = useAppStore((s) => s.todayDateKey());
  const ridesForDate = useAppStore((s) => s.ridesForDate);
  const todayRides = ridesForDate(todayDateKey);
  const remaining = Math.max(0, dailyGoal.targetNetProfit - todayNetProfit);
  const currency = useAppStore((s) => s.settings.currency);
  const progress =
    dailyGoal.targetNetProfit > 0
      ? Math.min(1, todayNetProfit / dailyGoal.targetNetProfit)
      : 0;

  if (!vehicle) {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.content}>
        <Card>
          <CardHeader
            title="Cadastre seu veículo"
            subtitle="Para calcular o lucro real das corridas."
          />
          <View style={styles.cardContent}>
            <Button
              fullWidth
              onPress={() => (navigation as any).navigate('Vehicle')}
            >
              Cadastrar veículo
            </Button>
          </View>
        </Card>
      </ScrollView>
    );
  }

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <View style={styles.section}>
        <Text style={styles.label}>Lucro líquido hoje</Text>
        <Text style={styles.bigAmount}>{formatCurrency(todayNetProfit, currency)}</Text>
        <Text style={styles.muted}>{todayRides.length} corrida(s) registrada(s)</Text>
      </View>

      <Card>
        <CardHeader
          title="Meta do dia"
          subtitle={`Meta: ${formatCurrency(dailyGoal.targetNetProfit, currency)}`}
        />
        <View style={styles.cardContent}>
          <View style={styles.progressBar}>
            <View
              style={[styles.progressFill, { width: `${progress * 100}%` }]}
            />
          </View>
          {remaining > 0 ? (
            <Text style={styles.muted}>Faltam {formatCurrency(remaining, currency)} para a meta</Text>
          ) : (
            <Text style={styles.success}>Meta atingida!</Text>
          )}
          <View style={styles.row}>
            <TouchableOpacity
              style={styles.halfButton}
              onPress={() => (navigation as any).navigate('Goal')}
            >
              <Button variant="secondary">Ver meta</Button>
            </TouchableOpacity>
            <TouchableOpacity
              style={styles.halfButton}
              onPress={() => (navigation as any).navigate('Ride')}
            >
              <Button fullWidth>Registrar corrida</Button>
            </TouchableOpacity>
          </View>
        </View>
      </Card>

      <View style={styles.row}>
        <TouchableOpacity
          style={styles.half}
          onPress={() => (navigation as any).navigate('WorthIt')}
        >
          <Card style={styles.quickCard}>
            <Text style={styles.quickIcon}>✓</Text>
            <Text style={styles.quickTitle}>Vale a pena?</Text>
            <Text style={styles.quickSub}>Antes de aceitar</Text>
          </Card>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.half}
          onPress={() => (navigation as any).navigate('History')}
        >
          <Card style={styles.quickCard}>
            <Text style={styles.quickIcon}>📋</Text>
            <Text style={styles.quickTitle}>Histórico</Text>
            <Text style={styles.quickSub}>Resumos e corridas</Text>
          </Card>
        </TouchableOpacity>
      </View>

      {todayRides.length > 0 && (
        <Card>
          <CardHeader title="Corridas de hoje" />
          {todayRides.slice(0, 5).map((r) => (
            <View key={r.id} style={styles.rideRow}>
              <Text style={styles.rideLabel}>
                {formatKm(r.distanceKm)} · {r.rideType === 'delivery' ? 'Entrega' : 'Passageiro'}
              </Text>
              <Text
                style={[
                  styles.rideValue,
                  r.netProfit >= 0 ? styles.positive : styles.negative,
                ]}
              >
                {formatCurrency(r.netProfit, currency)}
              </Text>
            </View>
          ))}
        </Card>
      )}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  section: { marginBottom: 24 },
  label: { fontSize: 14, color: '#64748b', marginBottom: 4 },
  bigAmount: { fontSize: 28, fontWeight: '700', color: '#0f172a' },
  muted: { fontSize: 14, color: '#64748b', marginTop: 4 },
  cardContent: { padding: 16 },
  progressBar: {
    height: 12,
    backgroundColor: '#e2e8f0',
    borderRadius: 6,
    overflow: 'hidden',
    marginBottom: 16,
  },
  progressFill: {
    height: '100%',
    backgroundColor: '#22c55e',
    borderRadius: 6,
  },
  success: { color: '#16a34a', fontWeight: '600', marginBottom: 16 },
  row: { flexDirection: 'row', gap: 12, marginBottom: 16 },
  halfButton: { flex: 1 },
  half: { flex: 1 },
  quickCard: { padding: 16, marginBottom: 0 },
  quickIcon: { fontSize: 24, marginBottom: 8 },
  quickTitle: { fontWeight: '600', color: '#0f172a', fontSize: 16 },
  quickSub: { fontSize: 12, color: '#64748b', marginTop: 4 },
  rideRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderTopWidth: 1,
    borderTopColor: '#f1f5f9',
  },
  rideLabel: { color: '#475569', fontSize: 15 },
  rideValue: { fontWeight: '600', fontSize: 15 },
  positive: { color: '#16a34a' },
  negative: { color: '#dc2626' },
});
