// apps-mobile/src/screens/RideInputScreen.tsx
import { useState } from 'react';
import { ScrollView, View, Text, TextInput, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useAppStore } from '../store/useAppStore';
import { formatCurrency } from '../lib/format';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';
import type { RideType } from '@driver-profit/shared';

export function RideInputScreen() {
  const navigation = useNavigation();
  const vehicle = useAppStore((s) => s.currentVehicle());
  const addRide = useAppStore((s) => s.addRide);
  const currency = useAppStore((s) => s.settings.currency);

  const [distance, setDistance] = useState('');
  const [payout, setPayout] = useState('');
  const [rideType, setRideType] = useState<RideType>('delivery');
  const [duration, setDuration] = useState('');
  const [lastAdded, setLastAdded] = useState<ReturnType<typeof addRide>>(null);

  const distanceNum = parseFloat(distance.replace(',', '.')) || 0;
  const payoutNum = parseFloat(payout.replace(',', '.')) || 0;
  const durationNum = duration ? parseFloat(duration.replace(',', '.')) || 0 : undefined;

  const handleSubmit = () => {
    if (!vehicle || distanceNum <= 0 || payoutNum <= 0) return;
    const record = addRide({
      distanceKm: distanceNum,
      payout: payoutNum,
      rideType,
      durationMinutes: durationNum,
    });
    setLastAdded(record);
    setDistance('');
    setPayout('');
    setDuration('');
  };

  const handleAddAnother = () => setLastAdded(null);

  if (!vehicle) {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.content}>
        <Card>
          <CardHeader title="Cadastre um veículo" subtitle="Vá em Veículo no menu." />
        </Card>
      </ScrollView>
    );
  }

  if (lastAdded) {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.content}>
        <Card>
          <View style={styles.success}>
            <Text style={styles.successTitle}>Corrida registrada!</Text>
            <Text style={styles.successSub}>
              Lucro líquido: {formatCurrency(lastAdded.netProfit, currency)}
            </Text>
            <View style={styles.successActions}>
              <Button variant="secondary" onPress={handleAddAnother} style={styles.half}>
                Nova corrida
              </Button>
              <Button onPress={() => (navigation as any).navigate('Home')} style={styles.half}>
                Início
              </Button>
            </View>
          </View>
        </Card>
      </ScrollView>
    );
  }

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <Card>
        <CardHeader
          title="Registrar corrida"
          subtitle="Dados da corrida ou entrega concluída."
        />
        <View style={styles.form}>
          <Text style={styles.label}>Tipo</Text>
          <View style={styles.typeRow}>
            <Button
              variant={rideType === 'delivery' ? 'primary' : 'secondary'}
              onPress={() => setRideType('delivery')}
              style={styles.typeBtn}
            >
              Entrega
            </Button>
            <Button
              variant={rideType === 'passenger' ? 'primary' : 'secondary'}
              onPress={() => setRideType('passenger')}
              style={styles.typeBtn}
            >
              Passageiro
            </Button>
          </View>
          <Text style={styles.label}>Distância (km)</Text>
          <TextInput
            style={styles.input}
            value={distance}
            onChangeText={setDistance}
            placeholder="Ex: 5,2"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Valor recebido</Text>
          <TextInput
            style={styles.input}
            value={payout}
            onChangeText={setPayout}
            placeholder="Ex: 18,50"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Tempo (min) — opcional</Text>
          <TextInput
            style={styles.input}
            value={duration}
            onChangeText={setDuration}
            placeholder="Ex: 25"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Button
            onPress={handleSubmit}
            disabled={distanceNum <= 0 || payoutNum <= 0}
            fullWidth
          >
            Salvar corrida
          </Button>
        </View>
      </Card>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  form: { padding: 16 },
  label: { fontSize: 14, fontWeight: '500', color: '#334155', marginBottom: 6 },
  input: {
    borderWidth: 1,
    borderColor: '#cbd5e1',
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 14,
    fontSize: 16,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
  typeRow: { flexDirection: 'row', gap: 12, marginBottom: 16 },
  typeBtn: { flex: 1 },
  success: { padding: 24, alignItems: 'center' },
  successTitle: { fontSize: 22, fontWeight: '700', color: '#16a34a', marginBottom: 8 },
  successSub: { fontSize: 16, color: '#475569', marginBottom: 24 },
  successActions: { flexDirection: 'row', gap: 12, width: '100%' },
  half: { flex: 1 },
});
