// apps-mobile/src/screens/SettingsScreen.tsx
import { useState } from 'react';
import { ScrollView, View, Text, TextInput, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useAppStore } from '../store/useAppStore';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';

export function SettingsScreen() {
  const navigation = useNavigation();
  const settings = useAppStore((s) => s.settings);
  const setSettings = useAppStore((s) => s.setSettings);
  const setDailyGoal = useAppStore((s) => s.setDailyGoal);
  const setWorthItThresholds = useAppStore((s) => s.setWorthItThresholds);

  const [goal, setGoal] = useState(String(settings.dailyGoal.targetNetProfit));
  const [minNet, setMinNet] = useState(String(settings.worthItThresholds.minNetProfit));
  const [minPerKm, setMinPerKm] = useState(String(settings.worthItThresholds.minProfitPerKm));
  const [minPerHour, setMinPerHour] = useState(String(settings.worthItThresholds.minProfitPerHour));

  const saveGoal = () => {
    const n = parseFloat(goal.replace(',', '.')) || 0;
    if (n >= 0) setDailyGoal({ ...settings.dailyGoal, targetNetProfit: n });
  };

  const saveThresholds = () => {
    setWorthItThresholds({
      minNetProfit: parseFloat(minNet.replace(',', '.')) || 0,
      minProfitPerKm: parseFloat(minPerKm.replace(',', '.')) || 0,
      minProfitPerHour: parseFloat(minPerHour.replace(',', '.')) || 0,
    });
  };

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <View style={styles.header}>
        <Button variant="ghost" onPress={() => navigation.goBack()}>
          ← Voltar
        </Button>
      </View>

      <Card>
        <CardHeader title="Meta diária" subtitle="Quanto você quer lucrar por dia (líquido)." />
        <View style={styles.form}>
          <Text style={styles.label}>Meta (valor)</Text>
          <TextInput
            style={styles.input}
            value={goal}
            onChangeText={setGoal}
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Button variant="secondary" onPress={saveGoal} fullWidth>
            Salvar meta
          </Button>
        </View>
      </Card>

      <Card>
        <CardHeader
          title="Vale a pena? — Mínimos"
          subtitle="Corrida só é 'vale a pena' se passar em todos."
        />
        <View style={styles.form}>
          <Text style={styles.label}>Lucro líquido mínimo</Text>
          <TextInput
            style={styles.input}
            value={minNet}
            onChangeText={setMinNet}
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Lucro mínimo por km</Text>
          <TextInput
            style={styles.input}
            value={minPerKm}
            onChangeText={setMinPerKm}
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Lucro mínimo por hora (0 = ignorar)</Text>
          <TextInput
            style={styles.input}
            value={minPerHour}
            onChangeText={setMinPerHour}
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Button variant="secondary" onPress={saveThresholds} fullWidth>
            Salvar mínimos
          </Button>
        </View>
      </Card>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  header: { marginBottom: 16 },
  form: { padding: 16 },
  label: { fontSize: 14, fontWeight: '500', color: '#334155', marginBottom: 6 },
  input: {
    borderWidth: 1,
    borderColor: '#cbd5e1',
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 14,
    fontSize: 16,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
});
