// apps-mobile/src/screens/VehicleScreen.tsx
import { useState } from 'react';
import {
  ScrollView,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Alert,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  type VehicleType,
  type FuelType,
  type VehicleProfile,
} from '@driver-profit/shared';
import { useAppStore } from '../store/useAppStore';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';

const VEHICLE_TYPES: { value: VehicleType; label: string }[] = [
  { value: 'car', label: 'Carro' },
  { value: 'motorcycle', label: 'Moto' },
  { value: 'bicycle', label: 'Bicicleta' },
];

const FUEL_TYPES: { value: FuelType; label: string }[] = [
  { value: 'gasoline', label: 'Gasolina' },
  { value: 'ethanol', label: 'Etanol' },
  { value: 'diesel', label: 'Diesel' },
  { value: 'cng', label: 'GNV' },
  { value: 'electric', label: 'Elétrico' },
  { value: 'none', label: 'Nenhum (bike)' },
];

export function VehicleScreen() {
  const navigation = useNavigation();
  const vehicles = useAppStore((s) => s.vehicles);
  const settings = useAppStore((s) => s.settings);
  const addVehicle = useAppStore((s) => s.addVehicle);
  const updateVehicle = useAppStore((s) => s.updateVehicle);
  const deleteVehicle = useAppStore((s) => s.deleteVehicle);
  const setSettings = useAppStore((s) => s.setSettings);

  const [editingId, setEditingId] = useState<string | null>(null);
  const [name, setName] = useState('');
  const [vehicleType, setVehicleType] = useState<VehicleType>('motorcycle');
  const [fuelType, setFuelType] = useState<FuelType>('gasoline');
  const [fuelPrice, setFuelPrice] = useState('');
  const [consumption, setConsumption] = useState('');
  const [maintenance, setMaintenance] = useState('');

  const resetForm = () => {
    setEditingId(null);
    setName('');
    setVehicleType('motorcycle');
    setFuelType('gasoline');
    setFuelPrice('');
    setConsumption('');
    setMaintenance('');
  };

  const startEdit = (v: VehicleProfile) => {
    setEditingId(v.id);
    setName(v.name);
    setVehicleType(v.vehicleType);
    setFuelType(v.fuelType);
    setFuelPrice(String(v.fuelPricePerLiter));
    setConsumption(String(v.consumptionKmPerLiter));
    setMaintenance(String(v.maintenanceCostPerKm));
  };

  const handleSubmit = () => {
    const fuelNum = parseFloat(fuelPrice.replace(',', '.')) || 0;
    const consNum = parseFloat(consumption.replace(',', '.')) || 0;
    const maintNum = parseFloat(maintenance.replace(',', '.')) || 0;

    if (editingId) {
      updateVehicle(editingId, {
        name: name || 'Veículo',
        vehicleType,
        fuelType,
        fuelPricePerLiter: fuelNum,
        consumptionKmPerLiter: consNum,
        maintenanceCostPerKm: maintNum,
      });
      resetForm();
      return;
    }

    addVehicle({
      name: name || 'Veículo',
      vehicleType,
      fuelType,
      fuelPricePerLiter: fuelNum,
      consumptionKmPerLiter: consNum,
      maintenanceCostPerKm: maintNum,
    });
    resetForm();
    if (vehicles.length === 0) navigation.goBack();
  };

  const handleDelete = (id: string) => {
    Alert.alert(
      'Remover veículo',
      'As corridas continuarão no histórico.',
      [
        { text: 'Cancelar', style: 'cancel' },
        { text: 'Remover', style: 'destructive', onPress: () => { deleteVehicle(id); resetForm(); } },
      ]
    );
  };

  const showFuel = fuelType !== 'none' && fuelType !== 'electric';

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()} style={styles.backBtn}>
          <Text style={styles.backText}>← Voltar</Text>
        </TouchableOpacity>
      </View>

      <Card>
        <CardHeader
          title={editingId ? 'Editar veículo' : 'Novo veículo'}
          subtitle="Consumo e custos para calcular o lucro real."
        />
        <View style={styles.form}>
          <Text style={styles.inputLabel}>Nome</Text>
          <TextInput
            style={styles.input}
            value={name}
            onChangeText={setName}
            placeholder="Ex: Moto iFood"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.inputLabel}>Tipo</Text>
          <View style={styles.optionsRow}>
            {VEHICLE_TYPES.map((t) => (
              <TouchableOpacity
                key={t.value}
                style={[styles.optionBtn, vehicleType === t.value && styles.optionBtnActive]}
                onPress={() => setVehicleType(t.value)}
              >
                <Text style={[styles.optionText, vehicleType === t.value && styles.optionTextActive]}>
                  {t.label}
                </Text>
              </TouchableOpacity>
            ))}
          </View>
          <Text style={styles.inputLabel}>Combustível</Text>
          <View style={styles.optionsRow}>
            {FUEL_TYPES.map((t) => (
              <TouchableOpacity
                key={t.value}
                style={[styles.optionBtn, fuelType === t.value && styles.optionBtnActive]}
                onPress={() => setFuelType(t.value)}
              >
                <Text style={[styles.optionText, fuelType === t.value && styles.optionTextActive]}>
                  {t.label}
                </Text>
              </TouchableOpacity>
            ))}
          </View>
          {showFuel && (
            <>
              <Text style={styles.inputLabel}>Preço do litro</Text>
              <TextInput
                style={styles.input}
                value={fuelPrice}
                onChangeText={setFuelPrice}
                placeholder="5,89"
                keyboardType="decimal-pad"
                placeholderTextColor="#94a3b8"
              />
              <Text style={styles.inputLabel}>Consumo (km/l)</Text>
              <TextInput
                style={styles.input}
                value={consumption}
                onChangeText={setConsumption}
                placeholder="30"
                keyboardType="decimal-pad"
                placeholderTextColor="#94a3b8"
              />
            </>
          )}
          <Text style={styles.inputLabel}>Custo manutenção por km</Text>
          <TextInput
            style={styles.input}
            value={maintenance}
            onChangeText={setMaintenance}
            placeholder="0,15"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <View style={styles.row}>
            {editingId && (
              <Button variant="ghost" onPress={resetForm} style={styles.half}>
                Cancelar
              </Button>
            )}
            <TouchableOpacity style={editingId ? styles.half : styles.full} onPress={handleSubmit}>
              <Button fullWidth>{editingId ? 'Salvar' : 'Cadastrar'}</Button>
            </TouchableOpacity>
          </View>
        </View>
      </Card>

      {vehicles.length > 0 && (
        <Card>
          <CardHeader title="Seus veículos" />
          {vehicles.map((v) => (
            <View key={v.id} style={styles.vehicleRow}>
              <View style={styles.vehicleInfo}>
                <Text style={styles.vehicleName}>{v.name}</Text>
                <Text style={styles.vehicleMeta}>
                  {VEHICLE_TYPES.find((t) => t.value === v.vehicleType)?.label} ·{' '}
                  {FUEL_TYPES.find((t) => t.value === v.fuelType)?.label}
                </Text>
              </View>
              <View style={styles.vehicleActions}>
                <TouchableOpacity onPress={() => setSettings({ vehicleId: v.id })}>
                  <Text style={[styles.actionText, settings.vehicleId === v.id && styles.active]}>
                    Usar
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => startEdit(v)}>
                  <Text style={styles.actionText}>Editar</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => handleDelete(v.id)}>
                  <Text style={[styles.actionText, styles.danger]}>Excluir</Text>
                </TouchableOpacity>
              </View>
            </View>
          ))}
        </Card>
      )}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  header: { flexDirection: 'row', marginBottom: 16 },
  backBtn: { padding: 8, minHeight: 44, justifyContent: 'center' },
  backText: { fontSize: 16, color: '#2563eb' },
  form: { padding: 16 },
  inputLabel: { fontSize: 14, fontWeight: '500', color: '#334155', marginBottom: 6 },
  input: {
    borderWidth: 1,
    borderColor: '#cbd5e1',
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 14,
    fontSize: 16,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
  row: { flexDirection: 'row', gap: 12 },
  half: { flex: 1 },
  full: { flex: 1 },
  vehicleRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderTopWidth: 1,
    borderTopColor: '#f1f5f9',
  },
  vehicleInfo: {},
  vehicleName: { fontWeight: '600', color: '#0f172a', fontSize: 16 },
  vehicleMeta: { fontSize: 14, color: '#64748b', marginTop: 2 },
  vehicleActions: { flexDirection: 'row', gap: 12 },
  actionText: { fontSize: 14, color: '#475569' },
  active: { color: '#2563eb', fontWeight: '600' },
  danger: { color: '#dc2626' },
  optionsRow: { flexDirection: 'row', flexWrap: 'wrap', gap: 8, marginBottom: 16 },
  optionBtn: {
    paddingHorizontal: 14,
    paddingVertical: 10,
    borderRadius: 10,
    backgroundColor: '#f1f5f9',
  },
  optionBtnActive: { backgroundColor: '#2563eb' },
  optionText: { fontSize: 14, color: '#475569' },
  optionTextActive: { color: '#fff', fontWeight: '600' },
});
