// apps-mobile/src/screens/WorthItScreen.tsx
import { useState } from 'react';
import { ScrollView, View, Text, TextInput, StyleSheet } from 'react-native';
import {
  calculateRideCost,
  evaluateWorthIt,
  type WorthItVerdict,
} from '@driver-profit/shared';
import { useAppStore } from '../store/useAppStore';
import { formatCurrency } from '../lib/format';
import { Card, CardHeader } from '../components/Card';
import { Button } from '../components/Button';

const VERDICT_STYLES: Record<WorthItVerdict, { label: string; emoji: string; color: string }> = {
  worth_it: { label: 'Vale a pena!', emoji: '✅', color: '#16a34a' },
  low_margin: { label: 'Margem baixa', emoji: '⚠️', color: '#d97706' },
  not_worth_it: { label: 'Não vale a pena', emoji: '❌', color: '#dc2626' },
};

export function WorthItScreen() {
  const vehicle = useAppStore((s) => s.currentVehicle());
  const thresholds = useAppStore((s) => s.settings.worthItThresholds);
  const currency = useAppStore((s) => s.settings.currency);

  const [distance, setDistance] = useState('');
  const [payout, setPayout] = useState('');
  const [duration, setDuration] = useState('');
  const [result, setResult] = useState<ReturnType<typeof calculateRideCost> | null>(null);
  const [verdict, setVerdict] = useState<WorthItVerdict | null>(null);

  const distanceNum = parseFloat(distance.replace(',', '.')) || 0;
  const payoutNum = parseFloat(payout.replace(',', '.')) || 0;
  const durationNum = duration ? parseFloat(duration.replace(',', '.')) || 0 : undefined;

  const handleCalculate = () => {
    if (!vehicle || distanceNum <= 0 || payoutNum <= 0) return;
    const breakdown = calculateRideCost(
      {
        distanceKm: distanceNum,
        payout: payoutNum,
        rideType: 'delivery',
        durationMinutes: durationNum,
      },
      vehicle
    );
    const v = evaluateWorthIt(breakdown, thresholds, durationNum);
    setResult(breakdown);
    setVerdict(v);
  };

  if (!vehicle) {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.content}>
        <Card>
          <CardHeader title="Cadastre um veículo" subtitle="Vá em Veículo no menu." />
        </Card>
      </ScrollView>
    );
  }

  const verdictInfo = verdict ? VERDICT_STYLES[verdict] : null;

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <Card>
        <CardHeader
          title="Vale a pena?"
          subtitle="Digite distância e valor antes de aceitar."
        />
        <View style={styles.form}>
          <Text style={styles.label}>Distância (km)</Text>
          <TextInput
            style={styles.input}
            value={distance}
            onChangeText={setDistance}
            placeholder="Ex: 5,2"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Valor da corrida</Text>
          <TextInput
            style={styles.input}
            value={payout}
            onChangeText={setPayout}
            placeholder="Ex: 15,00"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Text style={styles.label}>Tempo estimado (min) — opcional</Text>
          <TextInput
            style={styles.input}
            value={duration}
            onChangeText={setDuration}
            placeholder="Ex: 25"
            keyboardType="decimal-pad"
            placeholderTextColor="#94a3b8"
          />
          <Button
            onPress={handleCalculate}
            disabled={distanceNum <= 0 || payoutNum <= 0}
            fullWidth
          >
            Calcular
          </Button>
        </View>
      </Card>

      {result && verdictInfo && (
        <Card>
          <View style={styles.result}>
            <Text style={[styles.verdictEmoji, { color: verdictInfo.color }]}>
              {verdictInfo.emoji}
            </Text>
            <Text style={[styles.verdictLabel, { color: verdictInfo.color }]}>
              {verdictInfo.label}
            </Text>
            <View style={styles.breakdown}>
              <View style={styles.breakdownRow}>
                <Text style={styles.breakdownLabel}>Custo total</Text>
                <Text style={styles.breakdownValue}>
                  {formatCurrency(result.totalCost, currency)}
                </Text>
              </View>
              <View style={styles.breakdownRow}>
                <Text style={styles.breakdownLabel}>Lucro líquido</Text>
                <Text
                  style={[
                    styles.breakdownValue,
                    result.netProfit >= 0 ? styles.positive : styles.negative,
                  ]}
                >
                  {formatCurrency(result.netProfit, currency)}
                </Text>
              </View>
              <View style={styles.breakdownRow}>
                <Text style={styles.breakdownLabel}>Por km</Text>
                <Text style={styles.breakdownValue}>
                  {formatCurrency(result.profitPerKm, currency)}/km
                </Text>
              </View>
              {durationNum != null && durationNum > 0 && (
                <View style={styles.breakdownRow}>
                  <Text style={styles.breakdownLabel}>Por hora</Text>
                  <Text style={styles.breakdownValue}>
                    {formatCurrency(result.profitPerHour, currency)}/h
                  </Text>
                </View>
              )}
            </View>
          </View>
        </Card>
      )}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  content: { padding: 16, paddingBottom: 32 },
  form: { padding: 16 },
  label: { fontSize: 14, fontWeight: '500', color: '#334155', marginBottom: 6 },
  input: {
    borderWidth: 1,
    borderColor: '#cbd5e1',
    borderRadius: 12,
    paddingHorizontal: 16,
    paddingVertical: 14,
    fontSize: 16,
    marginBottom: 16,
    backgroundColor: '#fff',
  },
  result: { padding: 16 },
  verdictEmoji: { fontSize: 40, textAlign: 'center', marginBottom: 8 },
  verdictLabel: { fontSize: 20, fontWeight: '600', textAlign: 'center', marginBottom: 16 },
  breakdown: { gap: 8 },
  breakdownRow: { flexDirection: 'row', justifyContent: 'space-between' },
  breakdownLabel: { fontSize: 14, color: '#64748b' },
  breakdownValue: { fontSize: 14, fontWeight: '500' },
  positive: { color: '#16a34a' },
  negative: { color: '#dc2626' },
});
